synapse/replication/tcp/protocol.py

```diff
@@ -638,12 +648,16 @@ def connectionMade(self):
         # We've now caught up to position sent to us, notify handler.
         await self.handler.on_position(cmd.stream_name, cmd.token)
 
-        # When we get a `POSITION` command it means we've finished getting
-        # missing updates for the given stream, and are now up to date.
+        # We're now up to date wit the stream
         self.streams_connecting.discard(cmd.stream_name)
         if not self.streams_connecting:
             self.handler.finished_connecting()
 
+        # Check if the connection was closed underneath us, if so we bail
```

Comment:
are these temporary? they don't look very robust, since the connection could drop in the middle of any of the async function calls.

Reply:
Yeah, as per https://github.com/matrix-org/synapse/pull/7024#discussion_r396664851 we can be better in future. Though the checks should be fine since they protect `on_rdata` functions which are the ones that we want to be careful about.